lib/pushy_client/protocol_handler.rb

```diff
@@ -398,6 +407,21 @@ def check_server_address(name, server_address)
       end
     end
 
+    # Take the params and check to see if the stdout and stderr keys exceed the
+    # MAX_BODY_SIZE. If they do, log a warning and return the params without the
+    # stdout and stderr keys. If they do not, return the params hash unmodified.
+    def validate_params(params = {})
+      stdout_bytes = params[:stdout].to_s.bytesize
+      stderr_bytes = params[:stderr].to_s.bytesize
+
+      if (stdout_bytes + stderr_bytes) > MAX_BODY_SIZE
+        Chef::Log.warn("Command output too long. Will not be sent to server.")
```

Comment:
Out of curiosity - any particular reason why we drop both instead of just one if it individually happens to be less than MAX_BODY_SIZE...  i.e. still sending stdout if stderr is too large or something like that...

If it's "it was just simpler", that sounds fine :)